### PR TITLE
ABIFunctions: Split out a function for generating a comma separated list of variable names.

### PR DESCRIPTION
**Pull request by chriseth**
Part of https://github.com/ethereum/solidity/pull/6262/files

### CODE REVIEW

File: libsolidity/codegen/ABIFunctions.cpp [hunk 1]
```diff
@@ -90,14 +82,17 @@ string ABIFunctions::tupleEncoder(
 					<abiEncode>(<values> add(headStart, <pos>))
 				)")
 			);
-			elementTempl("values", valueNames);
+			string values = suffixedVariableNameList("value", stackPos, stackPos + sizeOnStack);
+			elementTempl("values", values.empty() ? "" : values + ", ");
```
Comment (chriseth) [inline libsolidity/codegen/ABIFunctions.cpp:86]: My change here: `","` -> `", "`

File: libsolidity/codegen/ABIFunctions.cpp [hunk 2]
```diff
@@ -636,29 +626,32 @@ string ABIFunctions::abiEncodeAndReturnUpdatedPosFunction(
 		_targetType.identifier() +
 		_options.toFunctionNameSuffix();
 	return createFunction(functionName, [&]() {
+		string values = suffixedVariableNameList("value", 0, numVariablesForType(_givenType, _options));
```
Comment (chriseth) [inline libsolidity/codegen/ABIFunctions.cpp:629]: My change: add `_options`

File: libsolidity/codegen/ABIFunctions.cpp [hunk 3]
```diff
@@ -1566,3 +1559,28 @@ size_t ABIFunctions::headSize(TypePointers const& _targetTypes)
...
+		result = _baseName + to_string(_endSuffix++);
+		while (_endSuffix < _startSuffix)
+			result = _baseName + to_string(_endSuffix++) + ", " + result;
+	}
+	return result;
+}
+
+size_t ABIFunctions::numVariablesForType(Type const& _type, EncodingOptions const& _options)
+{
+	if (!_options.encodeFunctionFromStack && _type.category() == Type::Category::Function)
```
Comment (chriseth) [inline libsolidity/codegen/ABIFunctions.cpp]: My change: Re-use `sizeOnStack` instead of computing it manually.

**Review by ekpyron — DISMISSED** [review]
Ok - I hadn't seen that ``_options.encodeFunctionFromStack`` would work to exclude the external function case - that's better then!

File: libsolidity/codegen/ABIFunctions.cpp [hunk 4]
```diff
@@ -1566,3 +1559,28 @@ size_t ABIFunctions::headSize(TypePointers const& _targetTypes)
...
+		return 1;
```
Comment (leonardoalt) [inline libsolidity/codegen/ABIFunctions.cpp:1583]: Reading the comment from the header file and this I'm confused if it should be 1 or 2.

**Comment by chriseth** [comment]
Updated.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**